frontend/src/App.tsx

```diff
@@ -35,61 +38,83 @@ const App = (): React.ReactElement => {
     currentUser,
   );
 
+  const [volunteerUser, setVolunteerUser] = useState<AuthenticatedVolunteer>(
+    null,
+  );
+
+  useEffect(() => {
```

Comment:
Because it relies on the client it would initially be a null value which might be harmful? I think we want the useEffect to be based on load of the app? Maybe I'm wrong here but when I set [] to be [authenticatedUser] it keeps reloading. I couldn't really try logging out because I keep getting this error ```{"error":"You are not authorized to make this request."}```

Reply:
maybe it's better to do this on the login page

Reply:
We should not havea. `useEffect` here on the `App.tsx` page. This essentially creates an almost infinite loop here and it's generally not the best practise to be setting state directly inside of the useEffect anyways.

The volunteer context should only be populated when a user logins and we see that their role equals volunteer.
You can set the volunteer context to some sort of default value blank for the app page.